greenlight/lib/tests/test_prefs.py

```diff
@@ -0,0 +1,150 @@
+
+        value = self.prefs.get_pref('browser.startup.homepage',
+                                    interface='nsIPrefLocalizedString')
+        self.assertNotEqual(value, properties_file)
+
+    def test_restore_pref(self):
+        # Test with single set_pref call and a new preference
+        self.prefs.set_pref(self.new_pref, True)
+        self.assertTrue(self.prefs.get_pref(self.new_pref))
+        self.prefs.restore_pref(self.new_pref)
+
+        orig_value = self.prefs.get_pref(self.string_pref)
+
+        # Test with single set_pref call
+        self.prefs.set_pref(self.string_pref, 'unittest')
+        self.assertEqual(self.prefs.get_pref(self.string_pref), 'unittest')
+        self.prefs.restore_pref(self.string_pref)
+        self.assertEqual(self.prefs.get_pref(self.string_pref), orig_value)
+
+        # Test with multiple set_pref call
+        self.prefs.set_pref(self.string_pref, 'unittest')
+        self.prefs.set_pref(self.string_pref, 'unittest2')
+        self.assertEqual(self.prefs.get_pref(self.string_pref), 'unittest2')
+        self.prefs.restore_pref(self.string_pref)
+        self.assertEqual(self.prefs.get_pref(self.string_pref), orig_value)
+
+        # Test with an unknown pref
+        self.assertRaises(MarionetteException,
+                          self.prefs.restore_pref, self.unknown_pref)
+
+    def test_restore_all_prefs(self):
+        orig_bool = self.prefs.get_pref(self.bool_pref)
+        orig_int = self.prefs.get_pref(self.int_pref)
+        orig_string = self.prefs.get_pref(self.string_pref)
+
+        self.prefs.set_pref(self.bool_pref, not orig_bool)
+        self.prefs.set_pref(self.int_pref, 99999)
+        self.prefs.set_pref(self.string_pref, 'unittest')
+
+        self.prefs.restore_all_prefs()
+        self.assertEqual(self.prefs.get_pref(self.bool_pref), orig_bool)
+        self.assertEqual(self.prefs.get_pref(self.int_pref), orig_int)
+        self.assertEqual(self.prefs.get_pref(self.string_pref), orig_string)
+
+    def test_set_pref_casted_values(self):
+        # basestring as boolean
+        self.prefs.set_pref(self.bool_pref, '')
+        self.assertFalse(self.prefs.get_pref(self.bool_pref))
+
+        self.prefs.set_pref(self.bool_pref, 'unittest')
+        self.assertTrue(self.prefs.get_pref(self.bool_pref))
+
+        # int as boolean
+        self.prefs.set_pref(self.bool_pref, 0)
+        self.assertFalse(self.prefs.get_pref(self.bool_pref))
+
+        self.prefs.set_pref(self.bool_pref, 5)
+        self.assertTrue(self.prefs.get_pref(self.bool_pref))
+
+        # boolean as int
+        self.prefs.set_pref(self.int_pref, False)
+        self.assertEqual(self.prefs.get_pref(self.int_pref), 0)
+
+        self.prefs.set_pref(self.int_pref, True)
+        self.assertEqual(self.prefs.get_pref(self.int_pref), 1)
+
+        # int as string
+        self.prefs.set_pref(self.string_pref, 54)
+        self.assertEqual(self.prefs.get_pref(self.string_pref), '54')
```

Comment:
I added some of those checks when we still had the `_cast()` method which actually returned the wrong type of value. So for basestring our `get_pref()` method returned an int.

Also with those tests I just wanted to make sure that we could cover changes in Firefox when our unit tests will be run on buildbot at one time. If we don't have them, we will only notice those things in our tests. Well, we should not do those strange things anyway, but I like safety. Tests are run quickly, and I don't see it would hurt. What do you think?